Migrated content from framework-libraries.md to class-library-overview.md (#35036)
dotnet/docs: docs/standard/class-library-overview.md

* <xref:System.Collections.Generic.List%601> - Represents a strongly typed list of objects that can be accessed by index. Is automatically resized as needed.
* <xref:System.Collections.Generic.Dictionary%602> - Represents a collection of values that are indexed by a key. Values can be accessed via key. Is automatically resized as needed.
* <xref:System.Uri> - Provides an object representation of a uniform resource identifier (URI) and easy access to the parts of the URI.
* <xref:System.DateTime> - Represents an instant in time, typically expressed as a date and time of day.

## Utility APIs

.NET includes a set of utility APIs that provide functionality for many important tasks.

* <xref:System.Net.Http.HttpClient> - An API for sending HTTP requests and receiving HTTP responses from a resource identified by a URI.
* <xref:System.Xml.Linq.XDocument> - An API for loading and querying XML documents with LINQ.
* <xref:System.IO.StreamReader> - An API for reading files.
* <xref:System.IO.StreamWriter> - An API for writing files.

## App-Model APIs

> There are many app-models that can be used with .NET, for example:

Reviewer: ```suggestion
There are many app models that can be used with .NET, for example:
```

> * [ASP.NET](https://dotnet.microsoft.com/apps/aspnet) - A web framework for building Web sites and services. Supported on Windows, Linux, and macOS (depends on ASP.NET version).

Reviewer: ```suggestion
* [ASP.NET](https://dotnet.microsoft.com/apps/aspnet) - A web framework for building web sites and services. Supported on Windows, Linux, and macOS (depends on ASP.NET version).
```

> * [Xamarin](https://dotnet.microsoft.com/apps/xamarin) - An app platform for building Android and iOS apps with .NET and C#. Supported on Windows and macOS.

Reviewer: ```suggestion
* [.NET MAUI](https://dotnet.microsoft.com/apps/maui) - An app platform for building native apps that run on Windows, macOS, iOS, and Android using C#.
```
